tests/scanner/scanners/external_project_access_scanner_test.py

```diff
@@ -0,0 +1,301 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for ExternalProjectAccessScanner."""
+# pylint: disable=line-too-long
+from datetime import datetime
+import json
+# pylint says unittest goes before mock
+import unittest
+import mock
+
+# pylint says sqlalchemy.orm goes before google.auth
+from sqlalchemy.orm import sessionmaker
+import google.auth
+from google.oauth2 import credentials
+
+from google.cloud.forseti.common.gcp_api import cloud_resource_manager as crm
+from google.cloud.forseti.common.gcp_api import api_helpers
+from google.cloud.forseti.common.gcp_type import resource_util
+from google.cloud.forseti.common.gcp_type.folder import Folder
+from google.cloud.forseti.common.gcp_type.organization import Organization
+from google.cloud.forseti.common.gcp_type.project import Project
+from google.cloud.forseti.common.gcp_type import resource as resource_mod
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.scanner.audit.external_project_access_rules_engine import Rule
+from google.cloud.forseti.scanner.scanners import external_project_access_scanner as epas
+from google.cloud.forseti.services.inventory.storage import initialize
+from google.cloud.forseti.services.inventory.storage import Storage
+from tests.common.gcp_api.test_data import fake_crm_responses
+from tests.common.gcp_api.test_data import http_mocks
+from tests.services.api_tests.inventory_test import TestServiceConfig
+from tests.services.inventory.storage_test import ResourceMock
+from tests.services.util.db import create_test_engine
+from tests.unittest_utils import ForsetiTestCase
+from tests.unittest_utils import get_datafile_path
+# pylint: enable=line-too-long
+
+TEST_EMAILS = ['user1@example.com', 'user2@example.com']
+
+LOGGER = logger.get_logger(__name__)
+
+
+class ExternalProjectAccessScannerTest(ForsetiTestCase):
+    """Tests for ExternalProjectAccessScanner"""
+
+    def setUp(self):
+        """Setup"""
+        self.global_configs = dict()
+        self.scanner_configs = (
+            dict(output_path='gs://test-forseti-dev/scanner/output',
+                 rules_path=__file__,
+                 scanners=[dict(name='external_project_access', enabled=True)]))
+
+        mock_invnetory_config = mock.MagicMock()
+        mock_invnetory_config.get_root_resource_id.return_value = (
+            'organizations/567890')
+
+        mock_service_config = mock.MagicMock()
+        mock_service_config.get_inventory_config.return_value = (
+            mock_invnetory_config)
+
+        self.service_config = mock_service_config
+        self.model_name = 'TestModel'
+        self.snapshot_timestamp = datetime.now().strftime('%Y%m%dT%H%M%SZ')
+        self.rules = get_datafile_path(
+            __file__, 'external_project_access_test_rules_1.yaml')
+
+        self.stash_email_method = epas.get_user_emails
+
+    def test_mock_emails(self):
```

Comment:
Why do we need to test getting emails from mocks?

Reply:
deleted